packages/devp2p/src/protocol/protocol.ts

```diff
@@ -0,0 +1,83 @@
+import ms from 'ms'
+import { debug as createDebugLogger } from 'debug'
+import { EventEmitter } from 'events'
+import { Peer, DISCONNECT_REASONS } from '../rlpx/peer'
+
+export class Protocol extends EventEmitter {
+  _peer: Peer
+  _statusTimeoutId: NodeJS.Timeout
+  _messageCodes: any
+  _debug: any
+  _verbose: any
```

Comment:
PR looks great! I'm about ready to approve and merge, but one last nit here maybe we can type these for added clarity?

My first pass writing it here `_messageCodes` would be something like `{ [key: string]: number }`, `_debug` you can try `createDebugLogger` or `ReturnType<createDebugLogger>`, and `_verbose` would be `boolean`?